plugins/workspace-plugin/src/workspace-projects-manager.ts

```diff
@@ -19,67 +19,63 @@ import * as theia from '@theia/plugin';
-  che.workspace.getCurrentWorkspace().then((workspace: cheApi.workspace.Workspace) => {
-    new WorkspaceProjectsManager(pluginContext, projectsRoot).run();
-  });
+  new WorkspaceProjectsManager(pluginContext, projectsRoot).run();
 }
 
 export class WorkspaceProjectsManager {
   protected watchers: theia.FileSystemWatcher[] = [];
   protected workspaceFolderUpdater = new WorkspaceFolderUpdater();
+  private outputChannel: theia.OutputChannel;
 
-  constructor(protected pluginContext: theia.PluginContext, protected projectsRoot: string) {}
+  constructor(protected pluginContext: theia.PluginContext, protected projectsRoot: string) {
+    this.outputChannel = theia.window.createOutputChannel('workspace-plugin');
+  }
 
-  getProjectPath(project: cheApi.workspace.devfile.Project): string {
+  getProjectPath(project: che.devfile.DevfileProject): string {
     return project.clonePath
       ? path.join(this.projectsRoot, project.clonePath)
-      : path.join(this.projectsRoot, project.name!);
-  }
-
-  getProjects(workspace: cheApi.workspace.Workspace): cheApi.workspace.devfile.Project[] {
-    const projects = workspace.devfile!.projects;
-    return projects ? projects : [];
+      : path.join(this.projectsRoot, project.name);
   }
 
   async run(): Promise<void> {
-    const workspace = await che.workspace.getCurrentWorkspace();
-    const cloneCommandList = await this.buildCloneCommands(workspace);
-
-    const cloningPromise = this.executeCloneCommands(cloneCommandList, workspace);
+    const devfile = await che.devfile.get();
+    this.outputChannel.appendLine(`Found devfile ${JSON.stringify(devfile, undefined, 2)}`);
+    const cloneCommandList = await this.buildCloneCommands(devfile.projects || []);
+    this.outputChannel.appendLine(`Clone commands are ${JSON.stringify(cloneCommandList, undefined, 2)}`);
+    const isMultiRoot = devfile.metadata?.attributes?.multiRoot === 'on';
+    this.outputChannel.appendLine(`multi root is ${isMultiRoot}`);
+    const cloningPromise = this.executeCloneCommands(cloneCommandList, isMultiRoot);
     theia.window.withProgress({ location: { viewId: 'explorer' } }, () => cloningPromise);
     await cloningPromise;
 
     await this.startSyncWorkspaceProjects();
   }
 
-  async buildCloneCommands(workspace: cheApi.workspace.Workspace): Promise<TheiaImportCommand[]> {
-    const instance = this;
+  notUndefined<T>(x: T | undefined): x is T {
+    return x !== undefined;
+  }
 
-    const projects = this.getProjects(workspace);
+  async buildCloneCommands(projects: che.devfile.DevfileProject[]): Promise<TheiaImportCommand[]> {
+    const instance = this;
 
     return projects
       .filter(project => !fs.existsSync(this.getProjectPath(project)))
-      .map(project => buildProjectImportCommand(project, instance.projectsRoot)!);
+      .map(project => buildProjectImportCommand(project, instance.projectsRoot))
+      .filter(command => this.notUndefined(command)) as TheiaImportCommand[];
```

Comment:
Would `command => command !== undefined` be working?

Reply:
it was to ensure it's providing correct type (and not undefined as return type)

it's because filter(command => command !== undefined) don't change return type as well

Reply:
https://github.com/microsoft/TypeScript/issues/5101 / https://github.com/microsoft/TypeScript/issues/20812